gpflow/conditionals/util.py

```diff
@@ -100,7 +113,15 @@ def base_conditional(Kmn: tf.Tensor,
     if not full_cov:
         fvar = tf.linalg.adjoint(fvar)  # [N, R]
 
-    return fmean, fvar  # [N, R], [R, N, N] or [N, R]
+    shape_constraints = [
+        (Kmn, [..., 'M', 'N']),  # for N
```

Comment:
This tensor got checked in the assert_shapes at the start of function already. I include it in here so I can get hold of the `N` dimension - but it could have been any other tensor that has an `N` somewhere in its dimensions. Same for `f` below.

Reply:
made the comment clearer